doc/specs/#1564 - Settings UI/spec.md

```diff
@@ -0,0 +1,123 @@
+---
+author: Kayla Cinnamon @cinnamon-msft
+created on: 2020-06-29
+last updated: 2020-07-08
+issue id: #1564
+---
+
+# Settings UI Implementation
+
+## Abstract
+
+This spec describes the basic functionality of the settings UI, including disabling it, the navigation items, launch methods, and editing of settings. The specific layout of each page will defined in later design reviews.
+
+## Inspiration
+
+We have been wanting a settings UI since the dawn of Terminal time, so we need to define how it will interact with the application and how users should expect to interact with it.
+
+## Solution Design
+
+The settings UI will be the default experience. We will provide users an option to skip the settings UI and edit the raw JSON file.
+
+### Ability to disable displaying the settings UI
+
+Some users don't want a UI for the settings. We can update the `openSettings` key binding with a `settingsUI` option.
+
+If people still like the UI but want to access the JSON file, we can provide an "Open the JSON file" button at the bottom of the navigation menu.
+
+### Launch methods
+
+#### Proposal 1 - Launch in a new window
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new window. This allows the user to edit their settings and see the Terminal live update with their changes.
+
+In the Windows taskbar, the icon will appear as if Terminal has multiple windows open.
+
+#### Proposal 2 - Launch in a new tab
+
+Clicking the settings button in the dropdown menu will open the settings UI in a new tab. This helps us take steps toward supporting non-terminal content in a tab. However, tab tear-off should be implemented before we have this. This way, users can still see their changes take effect if they want by tearing out the settings UI.
+
+### Editing and saving settings
+
+#### Proposal 1 - Automatically save settings
+
+As users edit fields in the settings UI, they are automatically saved and written to the JSON file. This allows the user to see their settings changes taking place in real time.
+
+#### Proposal 2 - Implement a save button
+
+Users will only see their settings changes take place once they click "Save". Clicking "Save" will write to the settings.json file. This aligns with the functionality that exists today by editing the settings.json file in a text editor and saving it.
+
+A future option could be adding a TerminalControl inside the settings UI to preview what the changes will look like before actually saving them to the settings.json file.
+

```

Comment:
Should we discuss at all the technical details of _how_ the settings are saved? Mention something about trying to make minimal updates to the user's settings file whenever possible? 

This might be more dev-spec'y, but I'm thinking also about things like deleting profiles. For user-defined profiles, we can just remove them from the json, easy. But for something like the default profiles and dynamic ones, we need to mark them as `hidden`.